codegen/src/test/resources/software/amazon/awssdk/codegen/poet/auth/scheme/query-auth-scheme-interceptor.java

```diff
@@ -70,13 +71,6 @@ private SelectedAuthScheme<? extends Identity> selectAuthScheme(List<AuthSchemeO
                 .getAttribute(SdkInternalExecutionAttribute.IDENTITY_PROVIDER_CONFIGURATION);
         List<Supplier<String>> discardedReasons = new ArrayList<>();
         for (AuthSchemeOption authOption : authOptions) {
-            if (authOption.schemeId().equals("smithy.api#noAuth")) {
-                if (!discardedReasons.isEmpty()) {
-                    LOG.debug(() -> String.format("%s auth will be used, discarded: '%s'", authOption.schemeId(),
-                            discardedReasons.stream().map(Supplier::get).collect(Collectors.joining(", "))));
-                }
-                return new SelectedAuthScheme<>(null, null, authOption);
-            }
```

Comment:
with this getting removed, we need to also configure the IdentityProviderConfiguration to include the IdentityProvider for the AnonymousIdentity in the SdkClientOption.IDENTITY_PROVIDER_CONFIGURATION that is configured for the client. Otherwise trySelectAuthScheme will return null and we will throw the "Failed to determine..." exception below.

Reply:
I think this might need the IdentityProvider in NoAuthScheme to not be private and moved into identity-spi too.

Reply:
see https://github.com/aws/aws-sdk-java-v2/blob/198a9d312be30d6fd0ced0c8afe9f2235ff937eb/core/sdk-core/src/main/java/software/amazon/awssdk/core/client/builder/SdkDefaultClientBuilder.java#L280

Reply:
I've been using the `acm` integration test as validation, which would probably fail with this PR. I think, we actually need a test in codegen-generated-classes-test that ensures the right auth scheme gets used. I don't think this PR needs to necessarily block on that though.

Reply:
The current implementation of the `identityProvider` returns an static one that does not take into account the configured ones, this works fine and it works pretty much the same way it was doing before.

```
    public IdentityProvider<AnonymousIdentity> identityProvider(IdentityProviderConfiguration providers) {
        return DEFAULT_IDENTITY_PROVIDER;
    }
```
FWIW, I just ran the `acm` integration test and it succeeded. Now, I understand that for completeness sake we might want to change this and select the anonymous identity if configured but I think that can be done in a follow up pull request, thoughts?

Reply:
Ah nice, I think this approach is good.